include/supl_os_client.rst

```diff
@@ -1,30 +1,206 @@
+
+SUPL Read
+   Callback function for receiving incoming data from the SUPL server.
+
+SUPL Write
+   Callback function for sending outgoing data to the SUPL server.
+
+
+SUPL session
+============
+
+The SUPL communication spans the following events / message transfers:
+
+* Receiving an event from the GPS module requesting to update A-GPS data
+* Executing subsequent message transfers until the completion of the A-GPS session
+* Pushing data into the GPS module
+
+It is assumed that the GNSS socket is already available.
+
+The following Message Sequence Chart (MSC) describes the flow of communication in a SUPL session.
+
+.. figure:: /images/supl_msc.svg
+   :alt: SUPL Session msc
+
+.. _supl_session_msc_desc:
+
+The various steps in the communication session are described below:
+
+1. The application receives a request from the GPS to start a SUPL session.
+   This request message (``NRF_GNSS_AGPS_DATA_ID``) is sent periodically from the GPS and received on the GNSS socket.
+
+#. If the LTE modem is deactivated the user must enable the LTE modem.
+   When modem is enabled it initiates a TCP socket to connect to the SUPL server.
+   In the SUPL session MSC above, the application opens a TCP socket to connect to the google SUPL server.
+#. The SUPL session is initiated by calling the :cpp:func:`supl_session` function with a copy of the ``nrf_gnss_agps_data_frame_t`` data that was received through the AGPS request event from the GPS module.
+   The SUPL client OS integration library generates the following parameters necessary for the session.
+
+   * ``supl_session_ctx_t`` structure from the AGPS request event data
+   * ``lte_params_t`` structure from the data read from the LTE modem
+   * ``device_id`` from IMSI number
+
+#. The SUPL client OS integration library then starts the SUPL session by calling the :cpp:func:`supl_client_session` function with the ``supl_session_ctx_t`` parameter.
+   The application does not return from this function until the SUPL session is finished or the :cpp:func:`supl_client_abort` function is called.
+#. The :cpp:func:`supl_client_session` function forms the entry point for the SUPL session.
+   There is no specific limit on the number of subsequent SUPL sessions that can be executed from this entry point.
+   See the documentation on the `SET-Initiated Non-Roaming Successful Case (Proxy mode)`_ for more information on the SUPL session.
+   The callback functions used for data transfer are listed below:
+
+    * SUPL Write (:cpp:func:`supl_write_t`) : callback for sending outgoing data to the SUPL server
+    * SUPL Read (:cpp:func:`supl_read_t`) : callback for receiving incoming data from the SUPL server
+
+#. The decoded SUPL data is sent to the GPS module using the AGPS Handler (:cpp:func:`agps_handler_t`) callback function.
+#. After the application returns from the :cpp:func:`supl_client_session` function, the TCP socket is no longer used by the SUPL client library and it can be closed.
```

Comment:
```suggestion
#. After the application returns from the :cpp:func:`supl_client_session` function, the TCP socket is no longer used by the SUPL client library and can be closed.
```
or
```suggestion
#. After the application returns from the :cpp:func:`supl_client_session` function, the TCP socket is no longer used by the SUPL client library, and it can be closed.
```